packages/ember-glimmer/lib/environment.js

```diff
@@ -68,7 +88,8 @@ export default class Environment extends GlimmerEnvironment {
         let definition = this.getComponentDefinition(path);
 
         if (definition) {
-          return new CurlyComponentSyntax({ args, definition, templates });
+          wrapClassAttribute(args);
```

Comment:
`args` is [EvaluatedArgs](https://github.com/tildeio/glimmer/blob/078009646b6838474aa2268f332721570153db8e/packages/glimmer-runtime/lib/compiled/expressions/args.ts#L32-L49) at this point and you do not want to special case any of this stuff in the rendering engine.